presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -617,6 +613,10 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
                     .collect(toList());
             checkState(!layouts.isEmpty(), "No usable layouts for %s", node);
 
+            if (layouts.stream().anyMatch(layout -> layout.getLayout().getPredicate().isNone())) {
+                return emptyRelation(node.getOutputSymbols());
```

Comment:
With `PickTableLayout` rule, is this line still reachable?

Reply:
Yes. `PickTableLayout` does not use `LayoutConstraintEvaluator`. See this commit changes the test below.